shell/platform/android/platform_view_android.cc

```diff
@@ -319,6 +319,15 @@ void PlatformViewAndroid::SetSemanticsEnabled(JNIEnv* env,
 
 void PlatformViewAndroid::ReleaseSurface() {
   NotifyDestroyed();
+
+  ftl::AutoResetWaitableEvent latch;
+  blink::Threads::Gpu()->PostTask([this, &latch]() {
+    if (surface_gl_->IsValid())
+      surface_gl_->GLContextClearCurrent();
+    latch.Signal();
+  });
+  latch.Wait();
+
   surface_gl_->TeardownOnScreenContext();
```

Comment:
surface_gl_ is created and primarily used by the Android main thread, so I'd prefer to call TeardownOnScreenContext() on that thread.  In particular, this ensures that TeardownOnScreenContext() has completed before we return back to Java.